### ISSUE TITLE
Quarkus shows warning messages for properties that are actually used but not part of config classes

### ISSUE THREAD

**Issue by geoand** [quarkusio/quarkus#5980]
**Describe the bug**
It seems like the big config PR removed he list of false positives keys that wouldn't trigger a warning when set.

**Expected behavior**
Setting `-Dquarkus.profile` (as is mentioned for example [here](https://quarkus.io/guides/config#configuration-profiles)) should not trigger the logging of any warning.

**Actual behavior**
Setting `-Dquarkus.profile` yields:

```
2019-12-06 11:48:32,749 WARN  [io.qua.config] (main) Unrecognized configuration key "quarkus.profile" was provided; it will be ignored
```

**To Reproduce**
1. Use any of the Quarkus quickstarts (from the `development` branch).
2. Build
3. Run with `-Dquarkus.profile` set

**Comment by geoand** [comment]
cc @dmlloyd

**Comment by gastaldi** [comment]
Duplicate of https://github.com/quarkusio/quarkus/issues/5855?

**Comment by geoand** [comment]
Yes it is. Although I'm pretty certain that it's not a `Good first issue` :), that is unless I missed something

**Comment by gastaldi** [comment]
According to @stuartwdouglas, the fix is to create a Config object for the missing properties, that's why I marked as a good first issue :)

**Comment by geoand** [comment]
That works for sure. But there are plenty of properties that need to be ignored: https://github.com/quarkusio/quarkus/blob/1.0.1.Final/core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigDefinition.java#L70

so I'm not sure if adding all those Config objects is the right way to go.

**Comment by gsmet** [comment]
I think it's OK to create a proper config for these. @gastaldi did you plan to work on it? I think we need it for CR so basically today.

If nobody's working on it, I'll do it.

**Comment by gastaldi** [comment]
> did you plan to work on it?

Nope, I'll leave the fun for you this time ;)

**Comment by geoand** [comment]
Having Config objects still won't solve the problem for `quarkus.profile` and `quarkus.version`

**Comment by gsmet** [comment]
@geoand why?

**Comment by geoand** [comment]
What Config object would you use to capture `quarkus.profile`? I guess there is some fundamental piece I am missing :)

**Comment by gsmet** [comment]
Hum. That's a good question.

Maybe a config root called `profile` and then a config item with a `ConfigItem#PARENT` name would work.

@dmlloyd is there a cleaner trick we could use? If at some point we want to add a `quarkus.locales` thing, it might be nice to have a proper config root with all these root properties.

**Comment by geoand** [comment]
The old exclude list also had `quarkus.version`

**Comment by machi1990** [comment]
The good thing with having a config root handling these false unknowns is that they'll be properly documented with the generated documentation.

**Comment by dmlloyd** [comment]
For root properties, either create a group and use the parent name like suggested above, or else we could enhance the config parser to support empty root names.  Either way isn't too hard.

**Comment by geoand** [comment]
The second solution seems more intuitive to me.

**Comment by dmlloyd** [comment]
I'll have a patch ready in moments ;)

**Comment by geoand** [comment]
:100:

**Comment by geoand** [comment]
Who wants to take this after #5989 is in? If no one wants to, I can do it
